bioconda_utils/lint_functions.py

```diff
@@ -85,268 +71,286 @@ def _has_preprocessing_selector(recipe):
             return True
 
 
-def in_other_channels(recipe, meta, df):
+def in_other_channels(recipe, metas, df):
     """
     Does the package exist in any other non-bioconda channels?
     """
-    results = _subset_df(recipe, meta, df)
-    channels = set(results.channel).difference(['bioconda'])
-    if len(channels):
-        return {
-            'exists_in_channels': channels,
-            'fix': 'consider deprecating',
-        }
+    for meta in metas:
```

Comment:
Regarding code complexity (or just cosmetics?):
Instead of adding loops explicitly, we could just add a decorator like
```python
def lint_multiple_metas(lint_function):  # better name?
    def lint_metas(recipe, metas, df, *args, **kwargs):
        lint = partial(lint_function, recipe)
        for meta in metas:
            ret = lint(meta, df, *args, **kwargs)
            if ret is not None:
                return ret
    return lint_metas

@lint_multiple_metas
def in_other_channels(recipe, metas, df):
[...]
```

Reply:
Yes, indeed, that's a good idea. But it should not stop us from merging. If nobody has the time for doing it, we can postpone it.

Reply:
done